test/e2e-go/features/stateproofs/stateproofs_test.go

```diff
@@ -739,3 +739,64 @@ func TestAttestorsChangeTest(t *testing.T) {
 
 	a.Equalf(int(consensusParams.StateProofInterval*expectedNumberOfStateProofs), int(lastStateProofBlock.Round()), "the expected last state proof block wasn't the one that was observed")
 }
+
+func TestSPWithTXPoolFull(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	defer fixtures.ShutdownSynchronizedTest(t)
+
+	t.Parallel()
+	a := require.New(fixtures.SynchronizedTest(t))
+
+	var fixture fixtures.RestClientFixture
+	configurableConsensus := make(config.ConsensusProtocols)
+	consensusVersion := protocol.ConsensusVersion("future")
+	consensusParams := getDefaultStateProofConsensusParams()
+	consensusParams.StateProofInterval = 4
+	configurableConsensus[consensusVersion] = consensusParams
+
+	fixture.SetConsensus(configurableConsensus)
+	fixture.SetupNoStart(t, filepath.Join("nettemplates", "TwoNodes50EachFuture.json"))
+
+	dir, err := fixture.GetNodeDir("Primary")
+	a.NoError(err)
+
+	cfg, err := config.LoadConfigFromDisk(dir)
+	a.NoError(err)
+	cfg.TxPoolSize = 0
+	cfg.SaveToDisk(dir)
+
+	dir, err = fixture.GetNodeDir("Node")
+	a.NoError(err)
+	cfg.SaveToDisk(dir)
+
+	fixture.Start()
+	defer fixture.Shutdown()
+
+	relay := fixture.GetLibGoalClientForNamedNode("Primary")
```

Comment:
verifyStateProofsCreation will attempt to send transactions, and the test will fail because the pool is full (has capacity 0).

Reply:
I don't understand something.
if the idea of this test is to demonstrate that stateproof can continue even if there is no space in the txn pool the tests should do the following:
config relay is a way that there is no space in txn pool
 do a sanity check for state proof i.e run verifyStateProofsCreation
 please help me understand what I am missing

Reply:
The test is doing exactly what you describe here. However, when is uses `verifyStateProofsCreation`, it will fail when the pool is full. That is why the function of `verifyStateProofsCreation` is done without using `verifyStateProofsCreation` and without sending payment transactions, or ignoring the error from sending transactions.